linkerd/proxy/http/src/normalize_uri.rs

```diff
@@ -36,6 +36,13 @@ pub struct NormalizeUri<S> {
     default: http::uri::Authority,
 }
 
+/// Detects the original form of a request URI and inserts a `WasAbsoluteForm`
+/// extension.
+#[derive(Clone, Debug)]
+pub struct DetectAbsoluteForm<S> {
```

Comment:
prefer something like `MarkAbsoluteForm`... We tend to use `Detect...` for connection-oriented things.

Reply:
naming this thing was probably one of the harder parts of the PR honestly :)